src/Security/Member.php

```diff
@@ -1702,4 +1710,46 @@ public function getHtmlEditorConfigForCMS()
         // If can't find a suitable editor, just default to cms
         return $currentName ? $currentName : 'cms';
     }
+
+    /**
+     * This needs to be public because it's referenced in getMemberPasswordField() and called in ConfirmedPasswordField
+     */
+    public static function generateRandomPassword(array $args = []): string
+    {
+        $password = '';
+        $validator = self::password_validator();
+        $minLength = $validator ? $validator->getMinLength() : 20;
+        $maxLength = $args['maxLength'] ?? 0;
+        if ($maxLength === 0) {
+            $maxLength = max($minLength, 20);
+        } else {
+            if ($maxLength < $minLength) {
+                throw new InvalidArgumentException('maxLength argument is less than PasswordValidator minLength');
+            }
+        }
+        $dummyMember = new Member();
+        // These are similar character sets that used in the default PasswordValidator
+        $lc = 'abcdefghijklmnopqrstuvwyxz';
+        $uc = strtoupper($lc);
+        $num = '0123456789';
+        $pun = '!@#$%^&*()_+-=[]{};:,./<>?';
+        $chars = $lc . $uc . $num . $pun;
+        $c = 0;
+        do {
```

Comment:
```suggestion
        $maxInt = strlen($chars) - 1;
        do {
```
No need to keep recreating this value for each character we add to the password.

Reply:
This wouldn't have any impact on performance since the while loop will usually only run once, not hundreds of thousands of times. I've updated the method to uses $charsets anyway, it also has strlen() in there though will keep it there.

Reply:
It's not about performance so much as it is about good coding conventions... but it's not worth arguing about so I'll leave it